packages/wasm-dpp/test/unit/dataContract/stateTransition/DataContractUpdateTransition/validation/state/validateDataContractUpdateTransitionStateFactory.spec.js

```diff
@@ -30,82 +39,87 @@ describe('validateDataContractUpdateTransitionStateFactory', () => {
 
     stateTransition = new DataContractUpdateTransition({
       dataContract: updatedRawDataContract,
+      protocolVersion: protocolVersion.latestVersion,
     });
 
     executionContext = new StateTransitionExecutionContext();
 
     stateTransition.setExecutionContext(executionContext);
 
-    stateRepositoryMock.fetchDataContract.resolves(dataContract);
+    const validator = new DataContractValidator();
+    const dataContractFactory = new DataContractFactory(protocolVersion.latestVersion, validator);
+    const wasmDataContract = await dataContractFactory.createFromBuffer(dataContract.toBuffer());
 
-    validateDataContractUpdateTransitionState = validateDataContractUpdateTransitionStateFactory(
-      stateRepositoryMock,
-    );
+    const stateRepositoryLike = {
+      fetchDataContract: () => wasmDataContract,
+    };
+
+    factory = (t) => validateDataContractUpdateTransitionState(stateRepositoryLike, t);
   });
 
   it('should return invalid result if Data Contract with specified contractId was not found', async () => {
-    stateRepositoryMock.fetchDataContract.resolves(undefined);
+    const stateRepositoryLikeNoDataContract = {
+      fetchDataContract: () => undefined,
+    };
+
+    const factoryNoDataContract = (t) => validateDataContractUpdateTransitionState(
```

Comment:
I think the name can be a little bit confusing for the next person working on it, I would call it something like `validateDataContractUpdateTransitionStateUndefinedContract` or something like that